instructionAPI/src/ArchSpecificFormatters.C

```diff
@@ -257,53 +250,16 @@ std::string x86Formatter::formatDeref(std::string addrString)
 
 std::string x86Formatter::getInstructionString(std::vector<std::string> operands) 
 {
-    // fprintf(stderr, "Operands: ");
-    // for(auto iter = operands.begin(); iter != operands.end(); iter++)
-    // {
-        // if(iter == operands.begin())
-            // fprintf(stderr, "%s", (*iter).c_str());
-        // else fprintf(stderr, ", %s", (*iter).c_str());
-    // }
-
-    /* We have to reorder the operands here */
-    std::string source_ops = "";
-    std::string dest_op = "";
-    std::string kmask_op = "";
-
-    /**
-     * We have to convert the Intel syntax operand ordering to AT&T because
-     * our tables are in Intel ordering
-     */
-    for(auto itr = operands.begin(); itr != operands.end(); itr++) 
-    {
-        std::string op = *itr;
-
-        /* If we still have a leading ##, it's an indirect call or SIB expression */
-        if(!op.compare(0, 2, "##"))
-        {
-            op = "0x0(" + op.substr(2) + ")";
-        }
-
-        if(itr == operands.begin())
-        {
-            dest_op = op;
-        } else if(!op.compare(0, 2, "%k"))
-        {
-            kmask_op = "{" + op + "}";
-        } else if(!source_ops.compare(""))
-        {
-            source_ops = op;
-        } else {
-            source_ops += "," + op;
-        }
+    std::string s;
```

Comment:
Could you add a quick comment with an example? On first pass, I missed the 'r' in `crbegin`.

Since we're in here fixing things, we might as well change the param to a `const&`. But I'll let you decide if you want to do that.

Reply:
I added a comment.  What type of example are looking for?

I was going to update the ArchSpecificFormatter interface to use const refs for vectors and strings for all the methods in a separate PR.

Reply:
I guess an example wouldn't really be helpful here.